tests/v2/validation/provisioning/rke2/ace_test.go

```diff
@@ -0,0 +1,217 @@
+		client    *rancher.Client
+	}{
+		{"Multiple Control Planes - Admin", nodeRoles0, r.client},
+		{"Multiple Control Planes - Standard", nodeRoles0, r.standardUserClient},
+	}
+
+	var name string
+	for _, tt := range tests {
+		subSession := r.session.NewSession()
+		defer subSession.Cleanup()
+
+		client, err := tt.client.WithSession(subSession)
+		require.NoError(r.T(), err)
+
+		cloudCredential, err := provider.CloudCredFunc(client)
+		require.NoError(r.T(), err)
+		for _, kubeVersion := range r.kubernetesVersions {
+			name = tt.name + providerName + " Kubernetes version: " + kubeVersion
+			for _, cni := range r.cnis {
+				name += " cni: " + cni
+				r.Run(name, func() {
+					testSession := session.NewSession(r.T())
+					defer testSession.Cleanup()
+
+					testSessionClient, err := tt.client.WithSession(testSession)
+					require.NoError(r.T(), err)
+
+					clusterName := namegen.AppendRandomString(provider.Name)
+					generatedPoolName := fmt.Sprintf("nc-%s-pool1-", clusterName)
+					machinePoolConfig := provider.MachinePoolFunc(generatedPoolName, namespace)
+
+					machineConfigResp, err := testSessionClient.Steve.SteveType(provider.MachineConfigPoolResourceSteveType).Create(machinePoolConfig)
+					require.NoError(r.T(), err)
+
+					machinePools := machinepools.RKEMachinePoolSetup(tt.nodeRoles, machineConfigResp)
+
+					cluster := clusters.NewK3SRKE2ClusterConfig(clusterName, namespace, cni, cloudCredential.ID, kubeVersion, machinePools)
+
+					cluster.Spec.LocalClusterAuthEndpoint = r.localClusterAuthEndpoint
+
+					clusterResp, err := clusters.CreateK3SRKE2Cluster(testSessionClient, cluster)
+					require.NoError(r.T(), err)
+
+					kubeProvisioningClient, err := r.client.GetKubeAPIProvisioningClient()
+					require.NoError(r.T(), err)
+
+					result, err := kubeProvisioningClient.Clusters(namespace).Watch(context.TODO(), metav1.ListOptions{
+						FieldSelector:  "metadata.name=" + clusterName,
+						TimeoutSeconds: &defaults.WatchTimeoutSeconds,
+					})
+					require.NoError(r.T(), err)
+
+					checkFunc := clusters.IsProvisioningClusterReady
+
+					err = wait.WatchWait(result, checkFunc)
+					assert.NoError(r.T(), err)
+					assert.Equal(r.T(), clusterName, clusterResp.ObjectMeta.Name)
+
+					clusterInfo, err := kubeProvisioningClient.Clusters(namespace).Get(context.TODO(), clusterResp.Name, metav1.GetOptions{})
+					require.NoError(r.T(), err)
+
+					clusterID := clusterInfo.Status.ClusterName
+					kubeConfig, err := kubeconfig.GetKubeconfig(client, clusterID) //client or r.client
+					require.NoError(r.T(), err)
+
+					original, err := r.client.SwitchContext(clusterResp.Name, kubeConfig)
+					require.NoError(r.T(), err)
+
+					originalResp, err := original.Resource(corev1.SchemeGroupVersion.WithResource("pods")).Namespace("").List(context.TODO(), metav1.ListOptions{})
+					require.NoError(r.T(), err)
+					for _, pod := range originalResp.Items {
+						r.T().Logf("Pod %v", pod.GetName())
+					}
```

Comment:
Just to make sure, below after the `GetContexts`, doesn't it include this `original` context as well? If it does, wouldn't it increase the readability if it was listing the pods with the same context once. If it doesn't, ignore this comment, please

Reply:
You are correct, however, the `kubeconfig` for these provisioned clusters has a default context that relates to the `original` context which will test/confirm access via `kubeconfig` prior to testing the context switching. The format is as follows:

```
contexts:
- name: "default-name"
  context:
    user: "default-name"
    cluster: "default-name"
- name: "context-1"
  context:
    user: "default-name"
    cluster: "context-1"
- name: "context-2"
  context:
    user: "default-name"
    cluster: "context-2"
- name: "context-3"
  context:
    user: "default-name"
    cluster: "context-3"

current-context: "default-name"
```